sources/packages/api/src/route-controllers/student/student.controller.ts

```diff
@@ -501,4 +503,54 @@ export class StudentController extends BaseController {
       restrictionMessage: studentRestrictionStatus.restrictionMessage,
     } as StudentRestrictionDTO;
   }
+
+  /**
+   * API to fetch student details by studentId.
+   * This API will be used by ministry users.
+   * @param studentId
+   * @returns Student Details
+   */
+  @AllowAuthorizedParty(AuthorizedParties.aest)
+  @Get("/aest/:studentId")
+  async getStudentDetails(
+    @Param("studentId") studentId: number,
+  ): Promise<StudentDetailDTO> {
+    const student = await this.studentService.findById(studentId);
+    const address = student.contactInfo.addresses[0];
+    const applications =
+      await this.applicationService.getAllStudentApplications(studentId);
+    const studentApplications = !applications
+      ? ([] as StudentApplicationSummary[])
```

Comment:
I believe that we could avoid the null check here and the ternary expression to make the code cleaner.
In the case there are no records, an empty array will be returned and the map will also be returning an empty array, right?

Reply:
Feedback comment addressed.